src/View/Components/ThemeToggle.php

```diff
@@ -27,32 +27,50 @@ public function render(): View|Closure|string
+
                                 document.documentElement.setAttribute('data-theme', this.theme)
-                                document.documentElement.setAttribute('class', this.theme)
                                 this.$dispatch('theme-changed', this.theme)
                             }
                         }"
                         @mary-toggle-theme.window="toggle()"
-                        {{ $attributes->class("swap swap-rotate") }}
+                        {{ $attributes->class(["swap swap-rotate", "focus-within:outline focus-within:outline-2 focus-within:outline-offset-2" => Str::contains($attributes->get('class'), "btn")]) }}
                     >
-                        <input type="checkbox" class="theme-controller opacity-0" @click="toggle()" :value="theme" />
+                        <input x-ref="input" type="checkbox" class="theme-controller opacity-0" @click="toggle()" value="dark" />
                         <x-mary-icon x-ref="sun" name="o-sun" class="swap-on" />
                         <x-mary-icon x-ref="moon" name="o-moon" class="swap-off"  />
                     </label>
                 </div>
                 <script>
-                    document.documentElement.setAttribute("data-theme", localStorage.getItem("mary-theme")?.replaceAll("\"", ""))
-                    document.documentElement.setAttribute("class", localStorage.getItem("mary-theme")?.replaceAll("\"", ""))
+                    // Set initial theme from three possible sources
+                    const savedTheme = localStorage.getItem("mary-theme")?.replaceAll("\"", "");
+                    // 1. saved theme from localStorage
+                    const definedTheme = document.documentElement.getAttribute("data-theme") || document.documentElement.classList.contains("dark") || null;
+                    // 2. defined theme from the <html>-tag
+                    const browserTheme = window.matchMedia && window.matchMedia('(prefers-color-scheme: dark)').matches ? "dark" : "light";
+                    // 3. browser theme/color-scheme
+                    // If none of the above would trigger and the browser doesn't support color-scheme it defaults to 'light'
+                    const endTheme = savedTheme ?? definedTheme ?? browserTheme;
+                    document.querySelector(".theme-controller").checked = endTheme === "dark"
+                    window.localStorage.setItem("mary-theme", JSON.stringify(endTheme))
+                    
+                    document.documentElement.setAttribute("data-theme", endTheme)
+                    if (endTheme === "dark") {
+                        document.documentElement.classList.add("dark")
+                        document.documentElement.classList.remove("light")
+                    } else {
+                        document.documentElement.classList.add("light")
+                        document.documentElement.classList.remove("dark")
+                    }
```

Comment:
As you navigate through routes it breaks

![image](https://github.com/robsontenorio/mary/assets/118955/6c9bf8ec-0df7-4b03-85e7-cc6bd355990d)

Reply:
Bug fixed, but these errors show's that your code also got executed on every page switch.